modules/lib/src/main/resources/assets/js/app/ui2/list/ContentIcon.tsx

```diff
@@ -24,61 +24,136 @@ import {
     FileMusic,
     Archive
 } from 'lucide-react';
-import {JSX} from 'react';
+import {useState, type JSX} from 'react';
+
+const ImgWithFallback = ({
+                             src,
+                             alt = '',
+                             fallback = <FileIcon/>,
+                             imgClassName,
+                         }: {
+    src?: string | null;
+    alt?: string;
+    fallback?: JSX.Element;
+    imgClassName?: string;
+}): JSX.Element => {
+    const [errored, setErrored] = useState(false);
+    if (!src || errored) {
+        return fallback;
+    }
+    return <img className={imgClassName} alt={alt} src={src} onError={() => setErrored(true)}/>;
+};
```

Comment:
Move this to a separate file and name it simply `<Image>`. This might become our default way of rendering images. There could be a lot going on here, but let’s keep it simple for now:

```tsx
<img
  className={className}
  alt={alt}
  width={size}        // optional
  height={size}       // optional
  loading={lazy ? 'lazy' : 'eager'} // eager by default
  decoding="async"
  referrerPolicy="no-referrer"
  src={src}
  srcSet={srcSet}     // omit for now, but we may add it later
  onError={handleError}
/>
```

`src` must always be set and treated as non-null.